File: Backend/Model/Challenge.cs
using System.Text.Json.Serialization;
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;

namespace AiRise.Models
{
    public class Challenge
    {
        [BsonId]
        [BsonRepresentation(BsonType.ObjectId)]
        public string? Id { get; set; }

        [BsonElement("name")]
        [JsonPropertyName("name")]
        public string Name { get; set; } = string.Empty;

        [BsonElement("description")]
        [JsonPropertyName("description")]
        public string Description { get; set; } = string.Empty;

        [BsonElement("url")]
        [JsonPropertyName("url")]
        public string Url { get; set; } = string.Empty;
    }
}

File: Frontend/composeApp/src/commonMain/kotlin/com/teamnotfound/airise/data/serializable/Challenge.kt
package com.teamnotfound.airise.data.serializable

import androidx.compose.runtime.mutableStateOf
import com.teamnotfound.airise.community.challenges.ChallengeUI
import kotlinx.serialization.Serializable

@Serializable
data class Challenge (
    val id: String? = null, // MongoDB ObjectId as String
    val name: String,
    val description: String,
    val url: String
) {
    fun toUI(): ChallengeUI = ChallengeUI(
        id = id ?: "",
        name = mutableStateOf(name),
        description = mutableStateOf(description),
        imageUrl = mutableStateOf(url)
    )
}